test/language/statements/async-function/evaluation-body-that-throws-after-await.js

```diff
@@ -0,0 +1,23 @@
+// Copyright 2016 Microsoft, Inc. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+
+/*---
+author: Brian Terlson <brian.terlson@microsoft.com>
+es7id: pending
+description: >
+  Errors thrown from the async function body reject the returned promise
+---*/
+
+async function foo() {
+  await Promise.resolve();
+  throw 1;
+}
+
+foo().then(function () {
+  $ERROR("Should not be called");
```

Comment:
You'll want to call `$DONE("Should not be called")` since [`$ERROR` simply throws an error](https://github.com/tc39/test262/blob/fbad34959ffedb02c5b8c040e0f992a6fc006f36/harness/sta.js#L16-L19), which in this context would reject the promise but not signal test failure.